src/components/Filters/SchemaSieve.ts

```diff
@@ -142,3 +149,136 @@ export const getOperators = (schema: JSONSchema6, field: string) => {
+	delimiter: string,
+	parentKey: string = '',
+	accumulator?: JSONSchema6,
+) => {
+	if (!accumulator) {
+		accumulator = cloneDeep(schema);
+
+		if (schema.properties) {
+			accumulator.properties = {};
+		}
+	}
+
+	if (schema.anyOf) {
+		accumulator.anyOf = schema.anyOf.map(subSchema =>
+			flattenAccumulator(subSchema, delimiter, parentKey),
+		);
+	}
+
+	forEach(schema.properties, (value: JSONSchema6, key) => {
+		const newKey = parentKey + DELIMITER + key;
+		// If the value is not an object type or
+		// If the value is a key value pair style object, it can be added to the
+		// accumulator
+		if (
+			value.type !== 'object' ||
+			(findKey(value.properties, { description: 'key' }) &&
+				findKey(value.properties, { description: 'value' }))
+		) {
+			// If there is a parentKey then this function has been called recursively
+			// and we should use the newKey
+			if (parentKey) {
+				accumulator!.properties![newKey] = defaults(value, { title: key });
+				return;
+			}
+
+			// If there is no parentKey then we are at the top level and this field
+			// does not need to be flattened/modified
+			accumulator!.properties![key] = value;
+			return;
+		}
+
+		accumulator = flattenAccumulator(value, delimiter, newKey, accumulator);
+	});
+
+	return accumulator;
+};
+
+// Reduces a multi level schema to a single level
+export const flattenSchema = (
+	schema: JSONSchema6,
+	delimiter: string = DELIMITER,
+): JSONSchema6 => {
+	return flattenAccumulator(schema, delimiter);
+};
+
+// Restores a schema that has been flattened with `flattenSchema()`
+export const unflattenSchema = (
+	schema: JSONSchema6,
+	delimiter: string = DELIMITER,
+) => {
+	const unflattenedSchema = cloneDeep(schema);
+
+	if (unflattenedSchema.anyOf) {
+		unflattenedSchema.anyOf = unflattenedSchema.anyOf.map(subSchema =>
+			unflattenSchema(subSchema, delimiter),
+		);
+	}
+
+	if (!unflattenedSchema.properties) {
+		return unflattenedSchema;
+	}
+
+	unflattenedSchema.properties = {};
+
+	forEach(schema.properties, (value: JSONSchema6, field) => {
+		// If the property hasn't been flattened, we can skip it
+		if (!startsWith(field, DELIMITER)) {
+			return (unflattenedSchema.properties![field] = value);
+		}
+
+		let head = unflattenedSchema;
+
+		const keys = trimStart(field, delimiter).split(delimiter);
+
+		while (keys.length) {
+			const key = keys.shift()!;
+
+			// Don't overwrite an existing entry
+			if (!head.properties![key]) {
+				head.properties![key!] = keys.length
+					? {
+							type: 'object',
+							properties: {},
+						}
+					: value;
+			}
+
+			head = head.properties![key] as JSONSchema6;
+		}
+	});
+
+	// Unflatten any required fields
+	if (unflattenedSchema.required) {
+		const requiredKeys = unflattenedSchema.required;
+
+		unflattenedSchema.required = [];
+
+		requiredKeys.forEach(requiredField => {
+			const keys = trimStart(requiredField, delimiter).split(delimiter);
+
+			let head = unflattenedSchema;
+
+			while (keys.length) {
```

Comment:
This could be a `for...of` and avoid the table manipulation, since we are not using `keys` inside the loop, but if you just prefer to keep it symmetric with the previous piece of code then :+1:

Reply:
Good point, I'll change the implementation